src/convert.rs

```diff
@@ -45,3 +45,35 @@ impl<T> fmt::Display for TryIntoError<T> {
 
 #[cfg(feature = "std")]
 impl<T: fmt::Debug> std::error::Error for TryIntoError<T> {}
+
+/// Error returned by the derived [`TryFrom`] implementation.
+///
+/// [`TryFrom`]: macro@crate::TryFrom
+#[derive(Clone, Copy, Debug)]
+pub struct TryFromError<T> {
+    /// Original input value which failed to convert via the derived
+    /// [`TryFrom`] implementation.
+    ///
+    /// [`TryFrom`]: macro@crate::TryFrom
+    pub input: T,
+}
+
+impl<T> TryFromError<T> {
+    #[doc(hidden)]
+    #[must_use]
+    #[inline]
+    pub const fn new(input: T) -> Self {
+        Self { input }
+    }
+}
+
+// `T` should only be an integer type and therefor display
+impl<T: fmt::Display> fmt::Display for TryFromError<T> {
```

Comment:
@ModProg that's too limiting bound. Usually, most of types don't impl `Display`, and thus we make the error here non-`Display`able for the majority of cases. Better remake the error message the way we lift this trait bound. I think adding a type name as `&'static str` to the error would do the trick, as it does for the `TryIntoError` above.

Reply:
Maybe it would make sense to have an explicit `TryFromRepr` error, then? Because all types available as repr do in fact implement `Display`.

Reply:
@ModProg yes, makes sense.